pkg/registration/spoke/spokeagent.go

```diff
@@ -505,16 +507,20 @@ func (o *SpokeAgentConfig) HasValidHubClientConfig(ctx context.Context) (bool, e
 // 2. The proxy url
 // 3. The CA bundle
 func (o *SpokeAgentConfig) isHubKubeconfigValid(ctx context.Context) (bool, error) {
-	bootstrapServer, bootstrapProxyURL, bootstrapCABndle, err := parseKubeconfig(o.registrationOption.BootstrapKubeconfig)
+	bootstrapCtxCluster, bootstrapServer, bootstrapProxyURL, bootstrapCABndle, err := parseKubeconfig(
+		o.registrationOption.BootstrapKubeconfig)
 	if err != nil {
 		return false, err
 	}
 
-	server, proxyURL, caBundle, err := parseKubeconfig(o.agentOptions.HubKubeconfigFile)
+	ctxCluster, server, proxyURL, caBundle, err := parseKubeconfig(o.agentOptions.HubKubeconfigFile)
 	switch {
 	case err != nil:
 		return false, err
-	case bootstrapServer != server, bootstrapProxyURL != proxyURL, !reflect.DeepEqual(bootstrapCABndle, caBundle):
+	case bootstrapCtxCluster != ctxCluster,
```

Comment:
we need some comment here or for the func on why we compare the clustername in the kubeconfig.